pkg/global/apply_configuration.go

```diff
@@ -24,33 +25,55 @@ import (
 // to all Buildbarn binaries, regardless of their purpose.
 func ApplyConfiguration(configuration *pb.Configuration) error {
 	// Push traces to Jaeger.
-	if jaegerConfiguration := configuration.GetJaeger(); jaegerConfiguration != nil {
-		pe, err := prometheus_exporter.NewExporter(prometheus_exporter.Options{
-			Registry:  prometheus.DefaultRegisterer.(*prometheus.Registry),
-			Namespace: "bb_storage",
-		})
-		if err != nil {
-			return util.StatusWrap(err, "Failed to create the Prometheus stats exporter")
-		}
-		view.RegisterExporter(pe)
+	if tracingConfiguration := configuration.GetTracing(); tracingConfiguration != nil {
 		if err := view.Register(ocgrpc.DefaultServerViews...); err != nil {
 			return util.StatusWrap(err, "Failed to register ocgrpc server views")
 		}
-		zpages.Handle(nil, "/debug")
-		je, err := jaeger.NewExporter(jaeger.Options{
-			AgentEndpoint:     jaegerConfiguration.AgentEndpoint,
-			CollectorEndpoint: jaegerConfiguration.CollectorEndpoint,
-			Process: jaeger.Process{
-				ServiceName: jaegerConfiguration.ServiceName,
-			},
-		})
-		if err != nil {
-			return util.StatusWrap(err, "Failed to create the Jaeger exporter")
+
+		if jaegerConfiguration := tracingConfiguration.Jaeger; jaegerConfiguration != nil {
+			je, err := jaeger.NewExporter(jaeger.Options{
+				AgentEndpoint:     jaegerConfiguration.AgentEndpoint,
+				CollectorEndpoint: jaegerConfiguration.CollectorEndpoint,
+				Process: jaeger.Process{
+					ServiceName: jaegerConfiguration.ServiceName,
+				},
+			})
+			if err != nil {
+				return util.StatusWrap(err, "Failed to create the Jaeger exporter")
+			}
+			trace.RegisterExporter(je)
+		}
+
+		if stackdriverConfiguration := tracingConfiguration.Stackdriver; stackdriverConfiguration != nil {
+			se, err := stackdriver.NewExporter(stackdriver.Options{
+				ProjectID: stackdriverConfiguration.ProjectId,
+				Location:  stackdriverConfiguration.Location,
+			})
+			if err != nil {
+				return util.StatusWrap(err, "Failed to create the Stackdriver exporter")
+			}
+			trace.RegisterExporter(se)
 		}
-		trace.RegisterExporter(je)
-		if jaegerConfiguration.AlwaysSample {
+
+		if tracingConfiguration.ExportPrometheus {
+			pe, err := prometheus_exporter.NewExporter(prometheus_exporter.Options{
+				Registry:  prometheus.DefaultRegisterer.(*prometheus.Registry),
+				Namespace: "bb_storage",
+			})
+			if err != nil {
+				return util.StatusWrap(err, "Failed to create the Prometheus stats exporter")
+			}
+			view.RegisterExporter(pe)
+		}
+
+		if tracingConfiguration.EnableZpages {
```

Comment:
I think this functionality is even broken at this point in time, as we never expose all of /debug of the default mux. Shall we just kick this out of the tree for the time being?

Reply:
Yeah, I was just trying to not disrupt too much, but if it's not being served up, no point in preserving it...

Reply:
Done